pkg/generator/secret_template_reconciler.go

```diff
@@ -75,7 +103,21 @@ func (r *SecretTemplateReconciler) Reconcile(ctx context.Context, request reconc
 		return reconcile.Result{}, err
 	}
 
-	if secretTemplate.DeletionTimestamp != nil {
+	if secretTemplate.DeletionTimestamp.IsZero() {
+		if !controllerutil.ContainsFinalizer(&secretTemplate, finalizerName) {
```

Comment:
AddFinalizer will only add if it doesn't already exists so we can remove the contains check.

Reply:
I think we still need the contains check because we only want to update the resource if the finalizer has been added.